torch/_dynamo/variables/builder.py

```diff
@@ -1497,7 +1497,7 @@ def wrap_tensor(self, value: torch.Tensor):
 
         self.tx.output.input_source_to_var[source] = tensor_variable
         assert "tensor_dict" not in tensor_proxy.node.meta
-        tensor_proxy.node.meta["tensor_dict"] = value.__dict__.copy()
+        tensor_proxy.node.meta["tensor_dict"] = {k: None if isinstance(v, torch.Tensor) else v for k, v in value.__dict__.items()}
```

Comment:
two things from talking to Joel:

(1) we should probably be using weakrefs to hold onto the tensors (right now I just hackily null out any tensor attributes to avoid a leak, although I don't know if dynamo actually relies on them being populated here)

(2) this doesn't handle attributes with further nested that hold tensors.